app/services/hyrax/characterizer.rb

```diff
@@ -0,0 +1,19 @@
+module Hyrax
+  ##
+  # @abstract Determins which characterizer to run based on the file_set type
+  # allowing implementation of Valkyrie file_sets
+  class Characterizer
+    ##
+    # @param source: the object to run a characterizer on
+    #
+    # @return [#characterize]
+    def self.for(source:)
+      case source
+      when Hyrax::FileSetBehavior # ActiveFedora
```

Comment:
would this ever be the case?  FileSetBehavior is a module not a class. https://github.com/samvera/hyrax/blob/addb09f382feb86ab32f937d1c41feaf8d7a1537/app/models/concerns/hyrax/file_set_behavior.rb#L2

Reply:
That is the case when passing a `FileSet` and not a `Hyrax::FileSet`. We see the same style here: https://github.com/samvera/hyrax/blob/master/app/services/hyrax/visibility_propagator.rb#L12. We verified this when stepping through this.

Reply:
@jcoyne yeah. @aaron-collier and I verified this yesterday.

Reply:
@jcoyne see:

```ruby
2.5.5 :001 > file_set = FileSet.new
 => #<FileSet id: nil, head: [], tail: [], depositor: nil, title: [], date_uploaded: nil, date_modified: nil, alternative_title: [], label: nil, relative_path: nil, import_url: nil, resource_type: [], creator: [], contributor: [], description: [], abstract: [], keyword: [], license: [], rights_notes: [], rights_statement: [], access_right: [], publisher: [], date_created: [], subject: [], language: [], identifier: [], based_near: [], related_url: [], bibliographic_citation: [], source: [], access_control_id: nil, embargo_id: nil, lease_id: nil> 

2.5.5 :002 > Hyrax::FileSetBehavior === file_set
 => true 
```

Reply:
This seems really unclear and easy to violate this assumption.

Reply:
Could you use `ActiveFedora::Base` instead?

Reply:
i'm :+1: on using `ActiveFedora::Base`, with the idea that as long as we're in a position where we need to do type checks we should do the least strict type checks that will work.

in theory, folks can implement a valid AF File Set class that doesn't include `Hyrax::FileSetBehavior`, right?